### PR TITLE
Fix empty map sent as array in Json requests

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #872 

/cc @stayallive

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -21,6 +21,8 @@
   </coverage>
   <php>
     <ini name="error_reporting" value="-1"/>
+    <env name="AWS_SHARED_CREDENTIALS_FILE" value="/dev/null" />
```
Comment (jderusse) [inline phpunit.xml.dist:24]: Not related to that PR but tests are broken if your machine has a region configured in your Default AWS profile

**Comment by stayallive** [comment]
Wow, this would have taken me quite a bit more time to figure out!

Thanks for taking the time to make this!

File: src/Service/DynamoDb/tests/Unit/DynamoDbClientTest.php [hunk 2]
```diff
@@ -104,7 +104,7 @@ public function testDeleteItem(): void
 
         $input = new DeleteItemInput([
             'TableName' => 'Foobar',
-
+            'Key' => ['ID' => ['S' => 'foobar']],
```
Comment (Nyholm) [inline src/Service/DynamoDb/tests/Unit/DynamoDbClientTest.php:107]: Why is this needed?

**Review by Nyholm — COMMENTED** [review]
Thank you for this PR. 

@stayalive, it is just a small change in the generator =)

We still need a test for #872.

File: src/Service/DynamoDb/tests/Unit/DynamoDbClientTest.php (hunk 2 above)
Comment (jderusse) [reply src/Service/DynamoDb/tests/Unit/DynamoDbClientTest.php:107]: hehe... previous code had a "bug"

`Key` is required.. But because previously we initialized maps with an empty array (see https://github.com/async-aws/aws/pull/873/files#diff-324c911f8afbcd579bb76e189d1af7df134c2f519be2414f97d830934ec23588L119) it never triggered the test `if ($this->key === null) {throw Exception(...);}`

File: phpunit.xml.dist (hunk 1 above)
Comment (Nyholm) [reply phpunit.xml.dist:24]: That is correct. 

It is also the correct behavior. 

Another fix for this would be to set `AWS_REGION=us-east-1`.

File: src/Service/DynamoDb/tests/Unit/DynamoDbClientTest.php (hunk 2 above)
Comment (Nyholm) [reply src/Service/DynamoDb/tests/Unit/DynamoDbClientTest.php:107]: Okey. Can you also update the changelogs with this?

**Comment by Nyholm** [comment]
The PHP8 build failure is not related

**Review by Nyholm — APPROVED** [review]
I think this PR looks good. 

But I really really do want a test for the bug @stayalive mentioned.

**Comment by Nyholm** [comment]
Thank you

**Comment by stayallive** [comment]
Sorry for not being able to provide some tests to prove my issue (was a bit crazy at work) but I appreciate both of you for fixing my half-assed attempt to do so ❤️

**Comment by Nyholm** [comment]
Thank you. 

Could you verify that this fix solve your problem? If so, I'll tag a new release.

**Comment by stayallive** [comment]
Yes, just confirmed that `"async-aws/dynamo-db": "dev-master#a4808ec37d9da540f096465717324c8b2d393bdb"` works as intended 💪
